lib/simple_job/sqs_job_queue.rb

```diff
@@ -282,110 +297,114 @@ def get_milliseconds(time = Time.now.utc)
       (time.to_f * 1000).round
     end
 
-    def log_execution(successful, message, job_type, start_milliseconds)
-      if self.class.config[:cloud_watch_namespace]
-        timestamp = Time.now.utc.to_s
-        environment = self.class.config[:environment]
-
-        # localhost throws an error when calling Socket.gethostbyname, don't call it in dev & test
-        hostname = if %w[development test].include?(environment)
-                     Socket.gethostname
-                   else
-                     Socket.gethostbyname(Socket.gethostname).first
-                   end
-
-        message_dimensions = [
-          { 'Name' => 'Environment', 'Value' => environment },
-          { 'Name' => 'SQSQueueName', 'Value' => queue_name },
-          { 'Name' => 'Host', 'Value' => hostname }
-        ]
-
-        job_dimensions = message_dimensions + [
-          { 'Name' => 'JobType', 'Value' => job_type }
-        ]
-
-        metric_data = [
-          {
-            'MetricName' => 'MessageCheckCount',
-            'Timestamp' => timestamp,
-            'Unit' => 'Count',
-            'Value' => 1,
-            'Dimensions' => message_dimensions
-          },
-          {
-            'MetricName' => 'MessageReceivedCount',
-            'Timestamp' => timestamp,
-            'Unit' => 'Count',
-            'Value' => message ? 1 : 0,
-            'Dimensions' => message_dimensions
-          },
-          {
-            'MetricName' => 'MessageMissCount',
-            'Timestamp' => timestamp,
-            'Unit' => 'Count',
-            'Value' => message ? 0 : 1,
-            'Dimensions' => message_dimensions
-          }
-        ]
-
-        if message
-          now = get_milliseconds
+    # localhost throws an error when calling Socket.gethostbyname, don't call it in dev & test
+    def hostname
+      @hostname ||= if %w[development test].include?(self.class.config[:environment])
+                      Socket.gethostname
+                    else
+                      Socket.gethostbyname(Socket.gethostname).first
+                    end
+    end
 
+    def log_execution(successful, message, job_type, start_milliseconds)
+      return unless self.class.config[:cloud_watch_namespace]
+
+      timestamp = Time.now.utc.iso8601
+
+      message_dimensions = [
+        { name: 'Environment', value: self.class.config[:environment] },
+        { name: 'SQSQueueName', value: queue_name },
+        { name: 'Host', value: hostname }
+      ]
+
+      job_dimensions = message_dimensions + [
+        { name: 'JobType', value: job_type }
+      ]
+
+      metric_data = [
```

Comment:
this looks like just converting to new style hash syntax.

Reply:
v1 keys were camel case and strings i.e. `'MetricName'`. v2 keys are snake case and symbols i.e. `metric_name`

Reply:
ah, yes. thanks.